### PR TITLE
Implement all remaining RouteGenerators

### PR DESCRIPTION
**Pull request by nareddyt**
No production behavior change yet, code is not used.

- Add route generators for `AllowCors`, `MethodNotAllowed`, and `CatchAll`
- Rename pre-existing route generators: `DirectResponseCORSGenerator`, `DirectResponseHealthCheckGenerator`, `ProxyBackendGenerator`
- Modify factory function to only generate 1 `RouteGenerator` instead of a slice

Testing done:
- Minimal config change when plugged into current `route_generator.go` (follow-up CL)

Follow-up CLs:
- Support HTTP backend
- Support timeout/deadline propagation

### CODE REVIEW

File: src/go/configgenerator/routegen/proxy_cors.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	return "cors_routes"
+}
+
+// GenRouteConfig implements interface RouteGenerator.
+func (g *ProxyCORSGenerator) GenRouteConfig(filterGens []filtergen.FilterGenerator) ([]*routepb.Route, error) {
+	backendHTTPPatterns := g.ProxyBackendGenerator.AffectedHTTPPatterns()
+
+	var corsHTTPPatterns httppattern.MethodSlice
+	seenUriTemplatesInRoute := make(map[string]bool)
+	for _, httpPattern := range backendHTTPPatterns {
```
Comment (nareddyt) [inline src/go/configgenerator/routegen/proxy_cors.go:64]: FYI forked from `ServiceInfo::processHttpRule()`

File: src/go/configgenerator/routegen/deny_invalid_method.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+		httpPatterns = append(httpPatterns, gen.AffectedHTTPPatterns()...)
+	}
+
+	if err := httppattern.Sort(&httpPatterns); err != nil {
+		return nil, err
+	}
+
+	var methodNotAllowedRoutes []*routepb.Route
+	seenUriTemplatesInRoute := make(map[string]bool)
+	for _, httpPattern := range httpPatterns {
```
Comment (nareddyt) [inline src/go/configgenerator/routegen/deny_invalid_method.go:63]: FYI forked from `route_generator.go::makeHttpRouteMatchers` (decoupled from making backend route matchers)

**Review by TAOXUY — APPROVED** [review]

**Pull request merged**